### PR TITLE
Interests screen: replaces UI tests with screenshot tests

### PR DESCRIPTION
**Pull request by JoseAlcerreca**
This is a controversial one. The original instrumented tests were painstakingly checking for specific strings whereas the new screenshot tests are simple and verify a lot more.

We could add a @Ignore to the instrumented test in case we need the tests in the future, but I think we should go all in and have the goal to remove all `assertIsDisplayed`s if they're not testing any UI logic to reduce build and emulator times.

### CODE REVIEW

File: feature/interests/src/test/java/com/google/samples/apps/nowinandroid/interests/InterestsScreenScreenshotTests.kt [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    fun interestsScreen_empty() {
+        composeTestRule.captureMultiDevice("interestsScreen_empty") {
+            InterestsScreen(InterestsUiState.Empty)
+        }
+    }
+
+    @Test
+    fun interestsScreen_loading() {
+        composeTestRule.captureMultiDevice("interestsScreen_loading") {
+            InterestsScreen(InterestsUiState.Empty)
```
Comment (mmoczkowski) [inline feature/interests/src/test/java/com/google/samples/apps/nowinandroid/interests/InterestsScreenScreenshotTests.kt:62]:
```suggestion
            InterestsScreen(InterestsUiState.Loading)
```

**Review by mmoczkowski — CHANGES_REQUESTED** [review]

File: feature/interests/src/test/java/com/google/samples/apps/nowinandroid/interests/InterestsScreenScreenshotTests.kt (hunk 1 above)
Comment (JoseAlcerreca) [reply feature/interests/src/test/java/com/google/samples/apps/nowinandroid/interests/InterestsScreenScreenshotTests.kt]: Oops good catch!

**Comment by mmoczkowski** [comment]
Existing instrumented tests to be replaced are:

```
niaLoadingWheel_inTopics_whenScreenIsLoading_showLoading
interestsWithTopics_whenTopicsFollowed_showFollowedAndUnfollowedTopicsWithInfo
topicsEmpty_whenDataIsEmptyOccurs_thenShowEmptyScreen
```

It seems to me, that everything that was evaluated by those tests is checked by the new screenshot tests. This is fine to me. I like the simplicity and comprehensiveness of screenshot tests. However, instead of removing instrumented tests completely, can we think of an instrumented test that cannot be replaced by a screenshot test?

**Comment by JoseAlcerreca** [comment]
Good question! Yes, the navigation tests (NavigationTest.kt) have user actions and even though we could replace the assertions with screenshots, there would be a lot of them. 

There are also some tests that use APIs like `performScrollToNode`. This would be easy to do with Roborazzi, but we're trying to keep user actions out of screenshot tests for compatibility with the preview-based tool.

**Review by mmoczkowski — COMMENTED** [review]
I like it. Let's see what @dturner has to say about this.

**Comment by dturner** [comment]
Can the screenshot tests be run locally, like instrumented tests? 

Running the instrumented tests locally is pretty important as it's far quicker at catching issues that way than solely relying on our painfully slow CI.

**Comment by JoseAlcerreca** [comment]
1- They can be run locally, but you should expect small anti-aliasing differences, especially on mac. This is annoying, but not a big deal once you understand why it happens.

2- Screenshot tests are the first thing that we run on CI, so you have results in ~15 mins iirc.

**Comment by keyboardsurfer** [comment]
Thank you for submitting this pull request. We have decided not to accept it at this time.
Feel free to comment or open an issue for further discussion if necessary.

**Pull request closed without merging**